newIDE/app/src/InstancesEditor/InstancesAdder.js

```diff
@@ -39,24 +42,52 @@ const roundPositionsToGrid = (
 export default class InstancesAdder {
   _instances: gdInitialInstancesContainer;
   _temporaryInstances: Array<gdInitialInstance>;
+  _onAdditionalServiceComplete: (infobarDetails: InfoBarDetails) => void;
   _options: Object;
   _zOrderFinder = new gd.HighestZOrderFinder();
 
-  constructor({ instances, options }: Props) {
+  constructor({ instances, options, onAdditionalServiceComplete }: Props) {
     this._instances = instances;
     this._options = options;
     this._temporaryInstances = [];
+    this._onAdditionalServiceComplete = onAdditionalServiceComplete;
   }
 
   setOptions(options: Object) {
     this._options = options;
   }
 
+  onAddInstanceAdditionalservice = (
+    instance: gdInitialInstance,
+    project: gdProject,
+    layout: gdLayout
+  ) => {
+    const additionalService = ObjectsAdditionalService.getServices(
+      instance,
+      layout,
+      true
+    );
+    if (additionalService) {
+      additionalService.onInstanceAdded(instance, project, layout);
+      const infoBarDetails = additionalService.getInfoBarDetails(
+        'onInstanceAdded'
+      );
+      if (infoBarDetails) {
+        this._onAdditionalServiceComplete(infoBarDetails);
+      }
+    }
+  };
+
   /**
    * Immediately create new instance at the specified position
    * (specified in scene coordinates).
    */
-  addInstances = (pos: [number, number], objectNames: Array<string>) => {
+  addInstances = (
+    pos: [number, number],
+    objectNames: Array<string>,
+    project: gdProject,
+    layout: gdLayout
```

Comment:
Let's not pass the project nor the layout. If the parent needs them, it will have them. The InstancesAdder is not supposed to need them, so let's not pass them.